azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+package azurerm
+
+import (
+	"bytes"
+	"sort"
+	"strconv"
+	"time"
+
+	"fmt"
+
+	"net/http"
+
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/insights"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmAutoscaleSetting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutoscaleSettingCreateOrUpdate,
+		Read:   resourceArmAutoscaleSettingRead,
+		Update: resourceArmAutoscaleSettingCreateOrUpdate,
+		Delete: resourceArmAutoscaleSettingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": locationSchema(),
+			"profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 20, // https://msdn.microsoft.com/en-us/library/azure/dn931928.aspx
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
No, this "name" is more like a label for the profile, it can be changed without having to create a new profile. It has the same behavior on Azure portal.